pkg/auth/authproviders/oidc/backend_impl.go

```diff
@@ -170,7 +170,8 @@ func (p *backendImpl) RefreshURL() string {
 	return ""
 }
 
-func (p *backendImpl) verifyIDToken(ctx context.Context, rawIDToken string, nonceVerification nonceVerificationSetting) (*authproviders.AuthResponse, error) {
+func (p *backendImpl) verifyIDToken(ctx context.Context, rawIDToken string, rawAccessToken string,
+	nonceVerification nonceVerificationSetting) (*authproviders.AuthResponse, error) {
```

Comment:
The responsibilities of this function grew—again!—but the name remained. Taking into account the changes you make to `fetchUserInfo()` I suggest we reorg the functions for clarity and similarity in flow structure in the following way:

* `verifyIDToken()` becomes `authFromIDToken()` returning `AuthResponse` (this will require a comment that we might call to userinfo as well); it does token verification, claim extraction, and conversion to auth response
* `fetchUserInfo()` becomes `authFromUserInfo()` returning `AuthResponse`
* `verifyIDToken()` is re-introduced and focuses on token verification only
* `claimsFromUserInfo()` is pretty much your `fetchUserInfoClaims()`
* `claimsFromIDToken()` converts the token and optionally calls into `claimFromUserInfo()`

I think it makes sense to do that in a separate PR to simplify reviewind.